src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -10,30 +36,160 @@ interface SpawnHostModalProps {
 }
 export const SpawnHostModal: React.FC<SpawnHostModalProps> = ({
   visible,
-  onOk,
   onCancel,
-}) => (
-  <Modal
-    title={<H2>Spawn New Host</H2>}
-    visible={visible}
-    onOk={onOk}
-    onCancel={onCancel}
-    footer={[
-      <Button onClick={onCancel}>Cancel</Button>,
-      <Button
-        data-cy="spawn-host-button"
-        disabled={false}
-        onClick={() => undefined}
-        variant="danger"
-      >
-        Restart
-      </Button>,
-    ]}
-    width="50%"
-    wrapProps={{
-      "data-cy": "patch-restart-modal",
-    }}
-  >
-    Spawn Host modal
-  </Modal>
-);
+}) => {
+  const { data: distrosData, loading: distroLoading } = useQuery<
+    DistrosQuery,
+    DistrosQueryVariables
+  >(GET_DISTROS, {
+    variables: {
+      onlySpawnable: true,
+    },
+  });
+
+  const { data: awsData, loading: awsLoading } = useQuery<
+    AwsRegionsQuery,
+    AwsRegionsQueryVariables
+  >(GET_AWS_REGIONS);
+
+  const { data: publicKeysData, loading: publicKeyLoading } = useQuery<
+    GetMyPublicKeysQuery,
+    GetMyPublicKeysQueryVariables
+  >(GET_MY_PUBLIC_KEYS);
+
+  const [publicKeyState, setPublicKeyState] = useState<publicKeyStateType>({
+    publicKey: {
+      name: "",
+      key: "",
+    },
+    savePublicKey: false,
+  });
+  const [hostDetailsState, setHostDetailsState] = useState<
+    hostDetailsStateType
+  >({
+    hasUserDataScript: false,
+    userDataScript: "",
+    expiration: null,
+    noExpiration: false,
+  });
+  const [distro, setDistro] = useState("");
+  const [awsRegion, setAWSRegion] = useState("");
```

Comment:
this is a lot of hooks. i find it incredibly helpful for scanning code to add a brief comment above each, e.g. 

```
// QUERY AWS REGIONS

// QUERY PUBLIC KEYS

// HOST DETAILS STATE
```